overlord/devicestate/handlers.go

```diff
@@ -103,13 +103,13 @@ func (m *DeviceManager) doSetModel(t *state.Task, _ *tomb.Tomb) error {
 	}
 	for snapName, snapst := range snapStates {
 		// TODO: remove this type restriction once we remodel
-		//       kernels/gadgets and add tests that ensure
+		//       gadgets and add tests that ensure
 		//       that the required flag is properly set/unset
 		typ, err := snapst.Type()
 		if err != nil {
 			return err
 		}
-		if typ != snap.TypeApp && typ != snap.TypeBase {
+		if typ != snap.TypeApp && typ != snap.TypeBase && typ != snap.TypeKernel {
```

Comment:
Shoul TypeSnapd be listed here (even if unused, so we don't forget?)

Reply:
This is a good question - in theory someone could go from a model with a snapd snap to a model without and then we could remove the snapd snap. But given our direction to introduce the snapd snap even on systems that only have "core" I think we actually want to keep the snapd snap even in this case.